File: king-admin-java/src/main/java/com/oukingtim/domain/SysRole.java
package com.oukingtim.domain;

import com.baomidou.mybatisplus.annotations.TableField;
import com.baomidou.mybatisplus.annotations.TableName;
import com.oukingtim.web.vm.JsTreeVM;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.util.List;

/**
 * Created by oukingtim
 */
@TableName("sys_role")
@Data
@EqualsAndHashCode(callSuper = false)
public class SysRole extends BaseModel<SysRole> {

    private String roleName;
    private String remark;
    @TableField(exist=false)
    private Boolean checked;
    @TableField(exist=false)
    private List<JsTreeVM> menuTree;

    public SysRole() {
    }

    public SysRole(String roleName) {
        this.roleName = roleName;
    }
}

File: king-admin-kotlin/src/main/kotlin/com/oukingtim/domain/SysRole.kt
package com.oukingtim.domain

import com.baomidou.mybatisplus.annotations.TableField
import com.baomidou.mybatisplus.annotations.TableName
import com.oukingtim.web.vm.JsTreeVM

/**
 * Created by oukingtim
 */
@TableName("sys_role")
class SysRole() : BaseModel<SysRole>() {

    var roleName: String? = null
    val remark: String? = null
    @TableField(exist = false)
    var checked: Boolean? = null
    @TableField(exist = false)
    val menuTree: List<JsTreeVM>? = null

    constructor(roleName: String):this(){
        this.roleName = roleName
    }
}